src/actions/hosts.ts

```diff
@@ -0,0 +1,33 @@
+import * as redux from 'redux';
+import { action } from 'typesafe-actions';
+import {
+  GET_HOSTS_REQUEST,
+  GET_HOSTS_SUCCESS,
+  GET_HOSTS_FAILURE
+} from '../constants/hosts';
+import * as hostsApi from '../api/hosts';
+import { Host } from '../models/hosts';
+
+// NOTE: this is just an example for now
+
+interface HostListApiResponse {
+  data: Host[];
+}
+
+export const getHosts = () => action(GET_HOSTS_REQUEST);
+export const getHostsSuccess = (data: HostListApiResponse) =>
+  action(GET_HOSTS_SUCCESS, data);
+
+export const getHostsFailure = () => action(GET_HOSTS_FAILURE);
```

Comment:
IIUC, these 3 action definitions could get simplified as:
```
const fetchHosts = createAsyncAction(
  GET_HOSTS_REQUEST,
  'GET_HOSTS_SUCCESS',
  'GET_HOSTS_FAILURE'
)<void, Host[], Error>();
```
then
```
const fetchHostsAsync = () => (dispatch: redux.Dispatch) => {
  dispatch(fetchHosts.request());
  getHosts()
    .then(response => {
      dispatch(fetchHosts.success((response.data)));
    })
    .catch(() => {
      dispatch(fetchHosts.failure(Error('some error)));
    });
};
```

I find this better because of naming. In api modules, we could use HTTP verbs (getHosts, postHosts, getHost...) and selectively import it here rather than using *.

Reply:
Man, I remember looking at those examples, and it wasn't clicking for me. I get it now, and yes, this is good.